src/components/searchBar/Results/index.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  useCallback,
+  useState,
+} from 'react';
+
+import { PDLink } from '@components/pdLink';
+import { PDScrollArea } from '@components/pdScrollArea';
+import { cn } from '@utils/helpers';
+
+import type { IMappedBlockExtrinsic } from '@custom-types/block';
+
+interface IButtonProps {
+  children: React.ReactNode;
+  isActive: boolean;
+  type: string;
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+interface IResultProps {
+  results: {
+    blockNumber: number | null;
+    extrinsics: IMappedBlockExtrinsic[] | null;
+  };
+  handleOpenModal: (e: React.MouseEvent<HTMLDivElement>) => void;
+  type: string;
+  classNames?: string;
+}
+
+const Button = (props: IButtonProps) => {
+  const { children, isActive, type, onClick } = props;
+  return (
+    <button
+      data-filter-type={type}
+      onClick={onClick}
+      className={cn(
+        'px-2 py-2.5 font-body2-regular hover:border-dev-pink-500',
+        'text-dev-white-400 hover:text-dev-white-200 dark:text-dev-black-800 dark:hover:text-dev-black-1000',
+        'border-b-2 border-b-transparent hover:border-b-dev-pink-500',
+        'duration-300 ease-in',
+        isActive && 'border-b-dev-pink-500',
+      )}
+    >
+      {children}
+    </button>
+  );
+};
+
+export const Results = (props: IResultProps) => {
+  const { results, handleOpenModal, type, classNames } = props;
+  const { blockNumber, extrinsics } = results;
+  const [filter, setFilter] = useState<string>(type ?? '');
+
+  const filteredExtrinsics = filter === 'all' || filter === 'extrinsics' ? extrinsics : null;
+  const showBlock = filter === 'all' || filter === 'block';
```

Comment:
these two things look like are supposed to do similar things, but achieve this in a different manner